File: src/test/resources/samples/c/ForbiddenControlFlow.c
#include <stdio.h>

void test_return(void)
{
    int condition = 1;

    if (condition) {
        return;
    }

    printf("This line will not be executed.\n");
}

int main(void)
{
    int i;

    for (i = 0; i < 10; i++) {
        if (i == 5) {
            break;
        }

        printf("%d\n", i);
    }

    for (i = 0; i < 10; i++) {
        if (i % 2 == 0) {
            continue;
        }

        printf("Odd: %d\n", i);
    }

    test_return();

    if (i == 10) {
        goto end;
    }

    printf("This line will be executed.\n");

end:
    printf("End of program.\n");

    return 0;
}

File: src/test/resources/samples/java/ForbiddenControlFlow.java
package samples;

public class ForbiddenControlFlow {

    public static void main(String[] args) {

        for (int i = 0; i < 10; i++) {
            if (i == 5) {
                break;
            }
            System.out.println(i);
        }

        for (int i = 0; i < 10; i++) {
            if (i % 2 == 0) {
                continue;
            }
            System.out.println("Odd: " + i);
        }

        Object obj = "Test";

        if (obj instanceof String) {
            System.out.println("Object is String.");
        }

        testReturn();
    }

    public static void testReturn() {

        boolean condition = true;

        if (condition) {
            return;
        }

        System.out.println("This line will not be executed.");
    }
}